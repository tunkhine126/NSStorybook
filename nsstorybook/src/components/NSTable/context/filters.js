import React, { useState, createContext } from 'react';
import PropTypes from 'prop-types';

export const NSFiltersContext = createContext();

export const NSFiltersProvider = ({
  children,
  filterCategories,
  lastUpdateProps,
  dateRangeProps,
  handleSelected,
}) => {
  const types = {
    multi: {},
    levels: {},
    single: {},
    date: {},
  };

  const { lastUpdate } = lastUpdateProps;

  ['multi', 'levels', 'single', 'date'].forEach(type => {
    filterCategories
      .filter(c => c.type === type)
      .forEach(({ id }) => {
        types[type][id] = [];
      });
  });

  const [singleSelectOptions, setSingleSelectOptions] = useState(types.single);
  const [multiSelectOptions, setMultiSelectOptions] = useState(types.multi);
  const [levelSelectOptions, setLevelSelectOptions] = useState(types.levels);
  const [showCalendar, setShowCalendar] = useState(
    lastUpdate.query === 'custom'
  );
  const [localDateRange, setlocalDateRange] = useState({
    before: null,
    after: null,
  });

  return (
    <NSFiltersContext.Provider
      value={{
        multiSelect: { multiSelectOptions, setMultiSelectOptions },
        levelSelect: { levelSelectOptions, setLevelSelectOptions },
        singleSelect: { singleSelectOptions, setSingleSelectOptions },
        calendarProps: { showCalendar, setShowCalendar },
        localDateProps: { localDateRange, setlocalDateRange },
        lastUpdateProps,
        dateRangeProps,
        filterCategories,
        handleSelected,
      }}
    >
      {children}
    </NSFiltersContext.Provider>
  );
};

NSFiltersProvider.propTypes = {
  children: PropTypes.any.isRequired,
  dateRangeProps: PropTypes.object.isRequired,
  lastUpdateProps: PropTypes.object.isRequired,
  filterCategories: PropTypes.array.isRequired,
  handleSelected: PropTypes.func.isRequired,
};
