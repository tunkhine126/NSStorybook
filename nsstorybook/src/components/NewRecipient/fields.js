/**
 *
 * Fields
 *
 */

import React, { useState } from 'react';
import PropTypes from 'prop-types';

import Grid from '@material-ui/core/Grid';

import NSFields from '../NSFields';

import { required } from 'utils/validations';
import { commonStyles } from './styles';

export function Fields({
  fields,
  showPreFixId,
  preFixSeparator,
  parentInternalId,
  preSetValue,
  fieldsRequired = false,
}) {
  let requiredProps = {};
  const classes = commonStyles();

  if (fieldsRequired) {
    requiredProps = {
      validateOnChange: true,
      validateOnBlur: true,
      required: true,
      validate: required,
    };
  }

  const [fieldState, setFieldState] = useState(
    fields.reduce((o, key) => ({ ...o, [key.name]: { item: [] } }), {})
  );

  const handleFieldChange = (field, value) =>
    setFieldState(state => ({ ...state, [field]: { item: value } }));

  const Prefix = () => (
    <p style={{ width: 'max-content' }}>{parentInternalId + preFixSeparator}</p>
  );

  const fieldsToRender =
    fields.length > 0 &&
    fields.map((f, idx) => {
      const selectFields = ['single_select', 'multi_select', 'likert'];
      if (selectFields.includes(f.field_type)) {
        return (
          <NSFields
            key={idx}
            fieldType={f.field_type}
            field={`${f.name}_custom`}
            label={f.contexts.us_en.label}
            values={fieldState[f.name]}
            options={f.option_values.map(o => ({ value: o }))}
            onchange={v => handleFieldChange(f.name, v)}
            {...requiredProps}
          />
        );
      }
      return (
        <NSFields
          key={idx}
          fieldType={f.field_type}
          field={`${f.name}_custom`}
          label={f.contexts.us_en.label}
          {...requiredProps}
        />
      );
    });
  return (
    <Grid item className={classes.spacing}>
      {fieldsRequired && (
        <NSFields
          fieldType="short_text"
          field="name"
          label="Name"
          {...requiredProps}
        />
      )}

      {!fieldsRequired && (
        <>
          <NSFields
            fieldType="photo"
            field="base64Avatar"
            label="Avatar"
            {...requiredProps}
          />

          <NSFields
            fieldType="short_text"
            field="internalId"
            label="Internal Id"
            position="start"
            initialValue={preSetValue}
            showPrefix={showPreFixId}
            icon={Prefix}
            {...requiredProps}
          />

          <NSFields
            fieldType="paragraph_text"
            field="description"
            label="Description"
            {...requiredProps}
          />
        </>
      )}

      {fieldsToRender}
    </Grid>
  );
}

Fields.propTypes = {
  fields: PropTypes.array.isRequired,
  fieldsRequired: PropTypes.bool,
  showPreFixId: PropTypes.bool,
  preFixSeparator: PropTypes.string,
  parentInternalId: PropTypes.string,
  preSetValue: PropTypes.string,
};

export default Fields;
