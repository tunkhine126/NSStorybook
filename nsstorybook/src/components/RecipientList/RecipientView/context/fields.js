import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

export const FieldsContext = createContext();

export const FieldsProvider = ({ children, ...rest }) => {
  const [activePhotoField, setActivePhotoField] = useState(null);
  const [recipientPhoto, setRecipientPhoto] = useState({});

  return (
    <FieldsContext.Provider
      value={{
        activePhotoProps: { activePhotoField, setActivePhotoField },
        recipientPhotoProps: { recipientPhoto, setRecipientPhoto },
        rest,
      }}
    >
      {children}
    </FieldsContext.Provider>
  );
};

FieldsProvider.propTypes = {
  children: PropTypes.any.isRequired,
};
