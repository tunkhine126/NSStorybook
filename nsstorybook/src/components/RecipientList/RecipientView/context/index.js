import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

export const RecipientViewContext = createContext();

export const RecipientViewProvider = ({
  children,
  tabChange,
  rdData,
  data,
  profileDetails,
  primaryRecipient,
  match,
  ...rest
}) => {
  const [toaster, setToaster] = useState({
    show: false,
    recipientUuid: null,
    field: null,
    undoHandler: null,
  });
  const [saveState, setSaveState] = useState({});

  return (
    <RecipientViewContext.Provider
      value={{
        data,
        rdData,
        tabChange,
        profileDetails,
        primaryRecipient,
        match,
        toasterProps: { toaster, setToaster },
        saveProps: { saveState, setSaveState },
        rest,
      }}
    >
      {children}
    </RecipientViewContext.Provider>
  );
};

RecipientViewProvider.propTypes = {
  children: PropTypes.any.isRequired,
  tabChange: PropTypes.func.isRequired,
  profileDetails: PropTypes.object.isRequired,
  primaryRecipient: PropTypes.bool.isRequired,
  data: PropTypes.object.isRequired,
  rdData: PropTypes.any.isRequired,
  match: PropTypes.any.isRequired,
};
