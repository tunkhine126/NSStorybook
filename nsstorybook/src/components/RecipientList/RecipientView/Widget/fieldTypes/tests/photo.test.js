/**
 *
 * Tests for Photo
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import TestingComponent from 'components/shared/Testing';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import Photo from '../photo';
import { FieldsProvider } from '../../../context/fields';

describe('<Photo />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <FieldsProvider>
          <Photo
            values={{ license_photo: '' }}
            field={{
              name: 'license_photo',
              contexts: {
                us_en: {
                  label: 'License Photo',
                },
              },
            }}
            label="License Photo"
            recipientData={{ photos: { license_photo: '' } }}
          />
        </FieldsProvider>
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <FieldsProvider>
          <Photo
            values={{ license_photo: '' }}
            field={{
              name: 'license_photo',
              contexts: {
                us_en: {
                  label: 'License Photo',
                },
              },
            }}
            label="License Photo"
            editMode
            recipientData={{ photos: { license_photo: '' } }}
          />
        </FieldsProvider>
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
