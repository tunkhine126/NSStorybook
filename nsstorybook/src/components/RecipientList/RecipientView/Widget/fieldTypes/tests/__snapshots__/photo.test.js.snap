// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Photo /> Should render and match the snapshot 1`] = `
<div
  class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <span
    class="MuiTypography-root makeStyles-caption-116 MuiTypography-caption MuiTypography-gutterBottom"
  >
    License Photo
  </span>
  <button
    class="MuiButtonBase-root MuiCardActionArea-root makeStyles-recipientChildImg-138"
    tabindex="0"
    type="button"
  >
    <img
      alt="License Photo"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
      height="100%"
      src="IMAGE_MOCK"
      title="License Photo"
    />
    <p
      class="makeStyles-imgActionText-139 img-action-text"
    >
      Click to change photo
    </p>
    <input
      accept="image/jpeg, image/png, image/tiff, image/ief, image/gif"
      type="file"
    />
    <span
      class="MuiCardActionArea-focusHighlight"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
